napari/view_layers.py

```diff
@@ -115,6 +116,7 @@ def view_image(
     viewer : :class:`napari.Viewer`
         The newly-created viewer.
     """
+    name = _guess_name(name, data)
```

Comment:
I dislike this line everywhere, because one cannot guess (heh) that guess-name is *not* guessing anything if name is None. My suggestion to remove the checking from the function and leave it to the caller:

```python
if name is None:
    name = _find_variable_name(data)
```

This would make these lines much clearer.